### PR TITLE
BigQuery: Documentation clarification for page_token

### LINKED ISSUE #3928: page_token parameter documentation is misleading
The `page_token` parameter to several api functions such as `list_projects` and `list_jobs` is documented as follows:
> opaque marker for the next "page" of projects. If not passed, the API will return the first page of projects.

This is very misleading, as these functions return an `HTTPIterator` which if iterated on will seamlessly return multiple pages of results.
Only though experimentation did I determine that multiple pages of results are returned and that if I did want to make use of `page_token` I'd need to access it at the `next_page_token` of the  `HTTPIterator`.

I think the documentation should be updated to better reflect that multiple pages can be fetched without touching the `page_token` and possibly include where to access `page_token` if someone does need to use it.

### PR DESCRIPTION
**Pull request by chemelnucfin**
Closes #3928 

Clarifies the page token parameter.

### CODE REVIEW

File: bigquery/google/cloud/bigquery/client.py [hunk 1]
```diff
@@ -138,9 +138,12 @@ def list_projects(self, max_results=None, page_token=None,
                             passed, defaults to a value set by the API.
 
         :type page_token: str
-        :param page_token: opaque marker for the next "page" of projects. If
-                           not passed, the API will return the first page of
-                           projects.
+        :param page_token: (Optional) Token representing a cursor into the
+                           projects.  If not passed, the API will return
```
Comment (tswast) [inline bigquery/google/cloud/bigquery/client.py]: Nit: I believe we use 1 space between sentences in our docs.

File: bigquery/google/cloud/bigquery/client.py [hunk 2]
```diff
@@ -138,9 +138,12 @@ def list_projects(self, max_results=None, page_token=None,
...
+                           the first page of projects. The token marks the
+                           beginning of the iterator to be returned and the
+                           value of the page_token can be accessed at
```
Comment (tswast) [inline bigquery/google/cloud/bigquery/client.py]: Use double backticks to show `page_token` in a code font.

File: bigquery/google/cloud/bigquery/client.py [hunk 3]
```diff
@@ -138,9 +138,12 @@ def list_projects(self, max_results=None, page_token=None,
...
+                           next_page_token of the HTTPiterator.
```
Comment (tswast) [inline bigquery/google/cloud/bigquery/client.py]: Ditto on backticks for ``next_page_token``.

Nit. `HTTPIterator`. Also, you could make this a link to the class.

File: bigquery/google/cloud/bigquery/client.py [hunk 4]
```diff
@@ -1229,7 +1241,11 @@ def list_rows(self, table, selected_fields=None, max_results=None,
 
         :type page_token: str
         :param page_token: (Optional) Token representing a cursor into the
-                           table's rows.
+                           table's rows.  If not passed, the API will return
+                           the first page of the rows. The token marks the
+                           beginning of the iterator to be returned and the
+                           value of the page_token can be accessed at
+                           next_page_token of the HTTPiterator.
```
Comment (tswast) [inline bigquery/google/cloud/bigquery/client.py]: `list_rows` uses a more specialized class than `HTTPIterator`. See the return type: 

```
:class:`~google.cloud.bigquery.table.RowIterator`
```

**Comment by chemelnucfin** [comment]
@tswast Thanks.

**Pull request merged**